### PR TITLE
[Add] Added modules membership_group and mass_mailing_membership_group modules

### PR DESCRIPTION
**Pull request by ByteMeAsap**

### CODE REVIEW

File: membership_group/models/membership_group.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class MembershipGroup(models.Model):
+    _name = "membership.group"
+    _description = "Membership Group"
+    _parent_store = True
+    _parent_name = "parent_id"
+
+    name = fields.Char()
+    membership_group_member_ids = fields.One2many("membership.group.member", "group_id")
+    parent_id = fields.Many2one(
+        comodel_name="membership.group", string="Parent Membership Group", index=True
```
Comment (tarteo) [inline membership_group/models/membership_group.py]: In the context of a membership group `string="Parent"` is enough

File: membership_group/models/membership_group.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+from odoo import api, fields, models
+
+
+class MembershipGroup(models.Model):
```
Comment (tarteo) [inline membership_group/models/membership_group.py]: Can we have a a display_name field which also includes the parents like so: `Parent / Child / Child of Child`

File: membership_group/menuitems.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        sequence="20"
+    />
+
+    <menuitem
+        id="membership_group_info_menu"
+        name="Membership Groups"
+        action="membership_group_action"
+        parent="membership_group.membership_group_top_menu"
+        sequence="1"
+    />
```
Comment (tarteo) [inline membership_group/menuitems.xml:17]: I think in the context of membership "Groups" is clear enough. IMO it's better to add this menu to Configuration.
The same for Membership Group Members but this one we can add to Reporting menu (maybe it's not possible as the menu currently is linked directly to a report.membership action) if we also add some pivot and graph.

Or maybe we can add a menu like this
* Groups
  * All
  * Reporting

File: membership_group/views/res_partner_view.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                    <field
+                        string="Membership Groups"
+                        name="membership_group_ids_count"
+                        widget="statinfo"
+                    />
+                </button>
+            </xpath>
+            <xpath expr="//field[@name='member_lines']" position="after">
+                <field name="membership_group_member_ids" nolabel="1" colspan="4">
+                    <tree editable="top">
```
Comment (tarteo) [inline membership_group/views/res_partner_view.xml]: We should add a label "Groups" (or something different) to make a separation now it's two lists after each other:
image

File: membership_group/models/membership_group_member.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from odoo import fields, models
+
+
+class MembershipGroupMember(models.Model):
+    _name = "membership.group.member"
+    _description = "Membership Group Member"
+
+    partner_id = fields.Many2one("res.partner", required=True, ondelete="cascade")
+    group_id = fields.Many2one("membership.group", required=True, ondelete="cascade")
```
Comment (tarteo) [inline membership_group/models/membership_group_member.py:9]: Now I can add group twice to a partner, should we constrain this (with _sql_constraints)?

image

File: membership_group/views/membership_group_view.xml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                            <field name="membership_group_member_ids">
+                                <tree editable="top">
+                                    <field
+                                        name="partner_id"
+                                        domain="[('membership_state', 'in', ['invoiced', 'paid', 'free'])]"
+                                    />
+                                </tree>
+                            </field>
+                        </page>
+                        <page string="Sub Membership Groups">
```
Comment (tarteo) [inline membership_group/views/membership_group_view.xml]: "Subgroups" would be fine I think

**Review by tarteo — COMMENTED** [review]
Great work!!

File: membership_group/models/membership_group.py (hunk 1 above)
Comment (ByteMeAsap) [reply membership_group/models/membership_group.py]: Updated

File: membership_group/models/membership_group.py (hunk 2 above)
Comment (ByteMeAsap) [reply membership_group/models/membership_group.py]: Added complete_name field

File: membership_group/menuitems.xml (hunk 3 above)
Comment (ByteMeAsap) [reply membership_group/menuitems.xml:17]: Added groups menu with All and reporting as its child and membership group members as child of reporting

File: membership_group/views/res_partner_view.xml (hunk 4 above)
Comment (ByteMeAsap) [reply membership_group/views/res_partner_view.xml]: Added a label 'Groups'

File: membership_group/models/membership_group_member.py (hunk 5 above)
Comment (ByteMeAsap) [reply membership_group/models/membership_group_member.py:9]: Added the sql constraint

File: membership_group/views/membership_group_view.xml (hunk 6 above)
Comment (ByteMeAsap) [reply membership_group/views/membership_group_view.xml]: Added

File: membership_hr/__manifest__.py [hunk 7]
```diff
@@ -10,13 +10,13 @@
     "website": "https://www.onestein.nl",
```
Comment (tarteo) [inline membership_hr/__manifest__.py:10]: Would be nice if this module is installed we have an applications menu in the membership menu, so people working with memberships don't have to switch to recruitement for it.

image

**Review by tarteo — APPROVED** [review]
Outside of the changes can you make allow_membership_registration default true, for at least the "main" website or add "default=True" . So we don't have to enable it after installation

**Comment by ByteMeAsap** [comment]
> Outside of the changes can you make allow_membership_registration default true, for at least the "main" website or add "default=True" . So we don't have to enable it after installation

Added default=True

File: membership_hr/__manifest__.py (hunk 7 above)
Comment (ByteMeAsap) [reply membership_hr/__manifest__.py:10]: Added applications menu in members->configuration if membership_hr_recruitment module is installed

**Pull request merged**